compiler/semcall.nim

```diff
@@ -657,7 +657,8 @@ proc searchForBorrowProc(c: PContext, startScope: PScope, fn: PSym): PSym =
       x = t.baseOfDistinct
     call.add(newNodeIT(nkEmpty, fn.info, x))
   if hasDistinct:
-    var resolved = semOverloadedCall(c, call, call, {fn.kind}, {})
+    let kind = if fn.kind == skProc: {skProc, skFunc} else: {fn.kind}
```

Comment:
I don't like this change, forward decls should be consistent.

Reply:
ok